compiler/src/dotty/tools/dotc/plugins/Plugin.scala

```diff
@@ -50,7 +51,19 @@ trait StandardPlugin extends Plugin {
    *  @param options commandline options to the plugin.
    *  @return a list of phases to be added to the phase plan
    */
-  def init(options: List[String]): List[PluginPhase]
+  @deprecated("`init` does not allow to access `Context`, use `initialize` instead.", since = "3.5.0")
+  def init(options: List[String]): List[PluginPhase] = Nil
```

Comment:
I wonder if we should keep it abstract? Otherwise, somebody might forget to override one of the variants. Would still make sense, as most of compiler plugins don't use context yet, so new variant can be treated as extended, advanced variant